components/net/image_cache.rs

```diff
@@ -390,21 +391,22 @@ impl ImageCache for ImageCacheImpl {
         }
     }
 
-    /// Return any available metadata or image for the given URL,
-    /// or an indication that the image is not yet available if it is in progress,
-    /// or else reserve a slot in the cache for the URL if the consumer can request images.
-    fn find_image_or_metadata(
+    fn get_image(&self, url: &ServoUrl, use_placeholder: UsePlaceholder) -> Option<Arc<Image>> {
+        let store = self.store.lock().unwrap();
+        match store.get_completed_image_if_available(url, use_placeholder) {
+            Some(Ok(ImageOrMetadataAvailable::ImageAvailable(x, _))) => Some(x),
+            _ => None,
+        }
+    }
+
+    fn track_image(
         &self,
         url: ServoUrl,
         use_placeholder: UsePlaceholder,
         can_request: CanRequestImages,
-    ) -> Result<ImageOrMetadataAvailable, ImageState> {
-        debug!("Find image or metadata for {}", url);
+    ) -> ImageCacheResult {
         let mut store = self.store.lock().unwrap();
-        if let Some(result) = store.get_completed_image_if_available(&url, use_placeholder) {
```

Comment:
If I'm reading this right, track_image now returns ImageCacheResult::LoadError if we ask the image cache to track a URL that has completed loading. That makes this API more challenging to use correct; I would rather continue supporting the case that we return completed images if they exist.

Reply:
I think I understand what you're pointing out. We only check the pending loads initially and just assume a load error if there's no entry instead of also checking the completed loads. So a call to `get_completed_image_if_available` should be made if it's not in the cache / `CacheResult::Miss(None)` so a complete load is checked for, right?

Reply:
I think there was code that used to do this in this method that was removed in this PR, so restoring it would be good.

Reply:
Right, you're commenting directly on the line and I completely missed it 🤦‍♂